src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h

```diff
@@ -1682,7 +1771,176 @@ namespace UiaOperationAbstraction
+        {
+        }
+
+        UiaTuple(TupleLocalType tuple) :
+            UiaTypeBase(std::make_shared<TupleLocalType>(std::move(tuple)))
+        {
+            ToRemote();
+        }
+
+        UiaTuple(const std::shared_ptr<TupleLocalType>& value) : UiaTypeBase(details::GetOrMakeSharedPtr(value))
+        {
+            ToRemote();
+        }
+
+        UiaTuple(typename ItemWrapperType::LocalType... args) : UiaTuple(TupleLocalType{ std::move(args)... })
+        {
+        }
+
+        UiaTuple(ItemWrapperType... args) : UiaTuple()
+        {
+            SetTupleWrapperItems<0 /* I */>(args...);
+        }
+
+        UiaTuple(winrt::Microsoft::UI::UIAutomation::AutomationRemoteArray array) :
+            UiaTypeBase(array)
+        {
+        }
+
+        explicit UiaTuple(winrt::Microsoft::UI::UIAutomation::AutomationRemoteAnyObject remoteValue) :
+            UiaTypeBase(remoteValue.AsArray())
+        {
+        }
+
+        UiaTuple(const UiaTuple&) = default;
+
+        UiaBool IsNull() const
+        {
+            if (ShouldUseRemoteApi())
+            {
+                auto remoteValue = std::get_if<RemoteType>(&m_member);
+                if (remoteValue)
+                {
+                    return remoteValue->IsNull();
+                }
+            }
+
+            // The local version of a UiaTuple can never be null.
+            return false;
+        }
+
+        UiaBool operator!() const { return IsNull(); }
+        operator UiaBool() const { return !IsNull(); }
+
+        operator std::shared_ptr<TupleLocalType>() const
+        {
+            return std::get<std::shared_ptr<TupleLocalType>>(m_member);
+        }
+
+        TupleLocalType& operator*()
+        {
+            return *std::get<LocalType>(m_member);
+        }
+
+        const TupleLocalType& operator*() const
+        {
+            return *std::get<LocalType>(m_member);
+        }
+
+        UiaBool operator==(const UiaTuple& rhs) const
+        {
+            if (ShouldUseRemoteApi())
+            {
+                auto mutableThis = *this;
+                mutableThis.ToRemote();
+
+                auto mutableRhs = rhs;
+                mutableRhs.ToRemote();
+                return std::get<RemoteType>(mutableThis.m_member).IsEqual(std::get<RemoteType>(mutableRhs.m_member));
+            }
+
+            auto localTuple = std::get<LocalType>(m_member);
+            auto localRhsTuple = std::get<LocalType>(rhs.m_member);
+            return *localTuple == *localRhsTuple;
+        }
+
+        UiaBool operator!=(const UiaTuple& rhs) const
+        {
+            return !(*this == rhs);
+        }
+
+        void ToRemote()
+        {
+            auto delegator = UiaOperationScope::GetCurrentDelegator();
+            if (delegator)
+            {
+                delegator->ConvertVariantDataToRemote<ItemWrapperType...>(m_member);
+            }
+        }
+
+        template <unsigned int I>
+        void SetAt(TupleItemWrapperType<I> item)

```

Comment:
Changing it to `Get<I>` and `Set<I>` could be less verbose...

Reply:
I'd somehow prefer `Get` and `Set`. The best way to articulate the preference would be that when I read `GetAt` I expect the index as a regular parameter, but the parameter is a non-type template param. I'm not sure if I've just looked at C++ for too long, though :) 

Anyway, no big deal if your or other folks' preference leans this way.

---
In reply to: [504466971](https://github.com/microsoft/Microsoft-UI-UIAutomation/pull/43#discussion_r504466971) [](ancestors = 504466971)

Reply:
My preference is towards `Get` and `Set`. And your reasons speak to me as well. Changed.

---
In reply to: [504998501](https://github.com/microsoft/Microsoft-UI-UIAutomation/pull/43#discussion_r504998501) [](ancestors = 504998501,504466971)

Reply:
I think I agree with this. I think ordinarily I'd be in favor of keeping this consistent with UiaArray, but also getting an element from a variable-length uniform-type array is fundamentally a different operation than getting one from a known-length heterogenous-type tuple, since the latter is more like struct field access. But also I don't have a super strong opinion.